processing/src/main/java/org/apache/druid/segment/transform/RowFunction.java

```diff
@@ -29,6 +30,8 @@
 public interface RowFunction
 {
   Object eval(Row row);
-
-  List<String> evalDimension(Row row);
+  default List<String> evalDimension(Row row)
```

Comment:
Add a test. And push the test upstream

Reply:
I think it would be cleaner to add the test for this upstream and once that PR is merged, cherry-pick it in a separate PR. Otherwise, there might be some changes suggested in the test and would require me to keep updating it in both places.

Reply:
Please create the test upstream then